src/prefect/environments/execution/remote/environment.py

```diff
@@ -0,0 +1,76 @@
+from typing import Any
+
+import cloudpickle
+
+from prefect.environments.execution import Environment
+from prefect.environments.storage import Docker
+from prefect.utilities import logging
+from prefect.utilities.configuration import set_temporary_config
+
+
+class RemoteEnvironment(Environment):
+    """
+    RemoteEnvironment is an environment which takes in information around an executor
+    and runs the flow in place using that executor.
+
+    Example using a RemoteEnvironment w/ an existing Dask cluster:
+    ```
+    RemoteEnvironment(
+        executor="prefect.engine.executors.DaskExecutor",
+        executor_kwargs={"address": "tcp://dask_scheduler_address"}
+    )
+    ```
+
+    Args:
+        - executor (str, optional): an importable string to an executor class; defaults
+            to `prefect.engine.executors.SynchronousExecutor`
+        - executor_kwargs (dict, optional): a dictionary of kwargs to be passed to
+            the executor; defaults to an empty dictionary
+    """
+
+    def __init__(
+        self,
+        executor: str = "prefect.engine.executors.SynchronousExecutor",
+        executor_kwargs: dict = dict(),
+    ) -> None:
+        self.executor = executor
+        self.executor_kwargs = executor_kwargs
+        self.logger = logging.get_logger("RemoteEnvironment")
+
+    def execute(  # type: ignore
+        self, storage: "Docker", flow_location: str, **kwargs: Any
+    ) -> None:
+        """
+        Run a flow from the `flow_location` here using the specified executor and
+        executor kwargs.
+
+        Args:
+            - storage (Docker): the Docker storage object that contains information relating
+                to the image which houses the flow
+            - flow_location (str): the location of the Flow to execute
+            - **kwargs (Any): additional keyword arguments to pass to the runner
+        """
+        if not isinstance(storage, Docker):
+            raise TypeError("RemoteEnvironment requires a Docker storage option")
```

Comment:
Could you log this as an error (or even critical?) before raising?

Reply:
Also, is this strictly necessary?  The below code will run with the [Local](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/environments/storage/local.py#L13) storage, for example